mg/target_test.go

```diff
@@ -0,0 +1,64 @@
+package mg
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+)
+
+func TestTarget(t *testing.T) {
+	t.Parallel()
+	dirs := []string{"testdata/src_dir", "testdata/target_dir"}
+	for _, d := range dirs {
+		os.MkdirAll(d, 0777)
+		time.Sleep(time.Second)
+	}
+	files := []string{
+		"testdata/src_dir/some_file",
+		"testdata/src_dir/some_",
+		"testdata/src_file",
+		"testdata/target_dir/some_file",
+		"testdata/target_file",
+	}
+	for _, v := range files {
+		err := ioutil.WriteFile(v, []byte(v), 0600)
+		if err != nil {
+			t.Fatal(err)
+		}
+		time.Sleep(time.Second)
+	}
+	defer func() {
+		err := os.RemoveAll("testdata")
```

Comment:
we may well have a permanent testdata dir in this directory, so can you please move these tests into a subdir of testdata, so that this won't remove permanent files we need at some point? :)